### ISSUE TITLE
Utilising resolve.modules for cleaner require paths

### ISSUE THREAD

**Issue by ChingxWando** [sysgears/apollo-universal-starter-kit#327]
Hey just wondering if we should start adding (__dirname)/src/client, /src/server to resolve.alias and refer to things like this:
```
import Routes from 'Client/app/routes';
import log from 'Common/log';
import settings from 'ProjRoot/settings';
```
Instead of:
```
import Routes from '../../client/app/routes';
import log from '../../common/log';
import settings from '../../../settings';
```

The obvious issue is that this breaks ssr. (other concerns?)
The problem is that system's (node's) resolve function do not recognise nor support webpack's module.alias feature. This is true even with the fact that our server code is transpiled with webpack and babel.

To enable node's resolve function people patch the resolve function with babel plugin:
[tleunen/babel-plugin-module-resolver](https://github.com/tleunen/babel-plugin-module-resolver)
[trayio/babel-plugin-webpack-alias](https://github.com/trayio/babel-plugin-webpack-alias)

Any thoughts if this is worth investigation?

**Comment by ChingxWando** [comment]
[trayio/babel-plugin-webpack-alias](https://github.com/trayio/babel-plugin-webpack-alias) is outdated and seemed hard to integrate anyways as it expects a standalone webpack config file.

The other babel plugin might be worth looking into. Just does relatively transparent patches to the require path like the webpack's alias thing does. Interesting from my experiments though, it eliminates the need to use webpack provide resolve.alias feature.

**Comment by vlasenko** [comment]
@ChingxWando I'm not sure why babel plugin is needed here. As you have said we are using Webpack to transpile both backend and each frontend code. We should not have any issues with using Webpack alias then. And I'm not sure why SSR will not work? I think everything should work.

**Comment by wesley-sw** [comment]
You are absolutely right, my early test failed due to a silly misconfiguration. The specifications of what happened is not really important - my alias entry just got overwritten by a existing entry in the baseConfig object.

This meant that we have been aware of it, there is actually one already in use... so do you think it is a good thing to have?

**Comment by wesley-sw** [comment]
We could leave it as is for stability of the boilerplate and let the users opt in themselves too. But I'm guessing we didn't have aliases enabled earlier simply because we were having to wait for the project structure to settle more.

Also, congratulating on the roll out of spinjs!

**Comment by vlasenko** [comment]
@wesley-sw It is a good thing to have if we will support common IDE's, like WebStorm, Atom, etc
